src/Environments/include/State.h

```diff
@@ -1,10 +1,32 @@
 #ifndef MCTS_LIBRARY_STATE_H
 #define MCTS_LIBRARY_STATE_H
 
-template <class T> class State {
-public:
-  State(T data) : m_data(data){};
-  T m_data;
+#include <any>
+#include <memory>
+
+struct Concept {
```

Comment:
What is a concept? Please use doxygen
https://github.com/Hoeiriis/MCTS#writing-documentation

Reply:
I'm uncertain to what extend it should be documented. It should only ever be used if anybody wants to extend the State class.